### DISCUSSION TITLE
Country support - yearly gain/loss

### DISCUSSION THREAD

**Discussion by JulesText** [eprbell/rp2#86] [Ideas]
An idea for the country infrastructure: in the US, tax reports seem to usually be done by the calendar year, and the fifo rp2 full report presents yearly gain/loss by calendar year. In other countries, the tax year can be different. Here in Australia, we file tax reports for the period July to June in the following year. In the UK its from 6 April to 5 April the following year.

The definition of year could be a country-specific feature. I appreciate this would require changes to the core computed_data.py script and the country-specify files. 

A further extension to this idea would be to add a new sheet in the country tax report template that aggregates the results from all of the transaction type sheets and provides the break down of that aggregate by tax year as net gain/loss.

Thank you for all the groundwork in developing this project :)

**Comment by eprbell** [comment]
That's interesting information: thanks for sharing. It will be relevant for #53, #51 and others. This will indeed affect some of the core logic, here are a few examples: 
* accounting_engine.py: `__years_2_methods`
* rp2_main.py: `years_2_accounting_method_names`
* computed_data.py: `YearlyGainLoss`
* more...
